ir/type.cpp

```diff
@@ -856,7 +856,27 @@ expr AggregateType::fromInt(expr v) const {
 }
 
 StateValue AggregateType::fromInt(StateValue v) const {
-  UNREACHABLE();
+  // structs can be empty
+  if (elements == 0)
+    return { expr::mkUInt(0, 1), true };
+
+  vector<StateValue> child_vals(elements);
+  unsigned bits = 0, np_bits = 0;
+  for (unsigned i = 0; i < elements; ++i) {
+    unsigned idx = little_endian ? i : elements - 1 - i;
+    auto *child = children[idx];
+    if (child->bits() == 0)
+      continue;
+
+    auto bits2 = bits + child->bits();
+    auto np_bits2 = np_bits + child->np_bits();
+    child_vals[idx]
+      = children[i]->fromInt({ v.value.extract(bits2 - 1, bits),
```

Comment:
AggregateType has an extract function. That should work I think, no?

Reply:
extract calls fromBV, so may not work in general.

Reply:
right. then maybe add a flag to extract to pick either fromBV/fromInt.
I really want to keep the number of functions that know about encoding minimal, so it's easier to experiment with different encodings. Also, it reduces opportunities for bugs to creep in.

Reply:
Updated